File: go/fiber/internal/modules/transaction/controller.go
package transaction

import (
	"tcc/internal/common/errors"
	"tcc/internal/common/helper"
	"tcc/internal/common/messages"
	"tcc/internal/modules/transaction/dto"

	"github.com/gofiber/fiber/v2"
)

type TransactionController struct{
	service TransactionService
}

func (c *TransactionController) FindAll(ctx *fiber.Ctx) error {
	idConta := ctx.Query("idConta")
	
	auth := helper.GetAuth(ctx)

	dto := dto.FindAllQueryTransactionsDto{
		IDConta: &idConta,
	}

	if err := helper.Validate(dto); err != nil {
		badRequest := errors.BadRequestError(messages.ErrorMessages.BadRequest, err...)
		return ctx.Status(badRequest.StatusCode).JSON(badRequest)
	}

	transactions, err := c.service.FindAll(*dto.IDConta, auth.ID, auth.Role)
	if err != nil {
		return ctx.Status(err.StatusCode).JSON(err)
	}

	return ctx.Status(fiber.StatusOK).JSON(transactions)
}

func (c *TransactionController) Create(ctx *fiber.Ctx) error {
	auth := helper.GetAuth(ctx)

	var dto dto.CreateTransactionDto
	if err := ctx.BodyParser(&dto); err != nil {
		badRequest := errors.BadRequestError(messages.ErrorMessages.BadRequest, err.Error())
		return ctx.Status(badRequest.StatusCode).JSON(badRequest)
	}
	
	if err := helper.Validate(dto); err != nil {
		badRequest := errors.BadRequestError(messages.ErrorMessages.BadRequest, err...)
		return ctx.Status(badRequest.StatusCode).JSON(badRequest)
	}

	_, err := c.service.Create(dto, auth.ID)
	if err != nil {
		return ctx.Status(err.StatusCode).JSON(err)
	}

	return ctx.Status(fiber.StatusCreated).Send(nil)
}

File: rust/axum/src/modules/transaction/controller.rs
use axum::{
    extract::{Extension, Query},
    http::StatusCode,
    response::IntoResponse,
    Json,
};
use validator::Validate;
use crate::modules::auth::interfaces::JwtPayload;
use crate::modules::transaction::service::TransactionService;
use crate::modules::transaction::dto::CreateTransactionDto;
use crate::modules::transaction::dto::FindAllQueryTransactionsDto;
use crate::common::errors::bad_request::bad_request_error;
use crate::common::messages::error::ErrorMessages;

pub struct TransactionController {
}

impl TransactionController {
    pub async fn find_all(
        Extension(auth): Extension<JwtPayload>,
        Query(query): Query<FindAllQueryTransactionsDto>,
    ) -> impl IntoResponse {       
        match TransactionService::find_all(query.id_conta, auth.id, auth.role).await {
            Ok(transactions) => (StatusCode::OK, Json(transactions)).into_response(),
            Err(err) => err.into_response(),
        }
    }

    pub async fn create(
        Extension(auth): Extension<JwtPayload>,
        Json(dto): Json<CreateTransactionDto>,
    ) -> impl IntoResponse {
        if let Err(err) = dto.validate() {
            return bad_request_error(ErrorMessages::BAD_REQUEST, Some(err)).into_response();
        }

        if let Err(err) = dto.validate_transaction() {
            return bad_request_error(ErrorMessages::BAD_REQUEST, Some(err)).into_response();
        }


        match TransactionService::create(dto, auth.id).await {
            Ok(_) => StatusCode::CREATED.into_response(),
            Err(err) => err.into_response(),
        }
    }
} 